app/api/bookyt/api/bookings.rb

```diff
@@ -29,6 +30,7 @@ class Bookings < Grape::API
               debit_account: debit_account,
               comments: params[:comments],
             }
+            attributes.merge!(reference: Invoice.find(params[:invoice_id])) if params[:invoice_id]
```

Comment:
This could result in an unhandled error if there is no Invoice with this ID. Would be creat if a proper error is returned in this case.